redbot/cogs/mod/mod.py

```diff
@@ -429,62 +448,111 @@ def __unload(self):
     @commands.command()
     @commands.guild_only()
     @checks.admin_or_permissions(ban_members=True)
-    async def hackban(self, ctx: commands.Context, user_id: int, *, reason: str = None):
-        """Preemptively bans user from the server
+    async def hackban(
+        self,
+        ctx: commands.Context,
+        user_ids: Greedy[RawUserIds],
+        days: Optional[DeletionDays] = 0,
+        *,
+        reason: str = None,
+    ):
+        """Preemptively bans user(s) from the server
 
-        A user ID needs to be provided in order to ban
+        User IDs need to be provided in order to ban
         using this command"""
+
+        banned = []
+        errors = {}
+
+        async def show_results():
+            text = f"Banned {len(banned)} users from the server."
+            if errors:
+                text += "\nErrors:\n"
+                text += "\n".join(errors.values())
+
+            for p in pagify(text):
+                await ctx.send(p)
+
+        def remove_processed(ids):
+            return [_id for _id in ids if _id not in banned and _id not in errors]
+
+        user_ids = list(set(user_ids))  # No dupes
+
         author = ctx.author
         guild = ctx.guild
+
+        if not user_ids:
+            await ctx.send_help()
+            return
+
         if not guild.me.guild_permissions.ban_members:
             return await ctx.send(_("I lack the permissions to do this."))
-        is_banned = False
+
         ban_list = await guild.bans()
         for entry in ban_list:
-            if entry.user.id == user_id:
-                is_banned = True
-                break
+            for user_id in user_ids:
+                if entry.user.id == user_id:
+                    errors[user_id] = f"User {user_id} is already banned."
+
+        user_ids = remove_processed(user_ids)
 
-        if is_banned:
-            await ctx.send(_("User is already banned."))
+        if not user_ids:
+            await show_results()
             return
 
-        user = guild.get_member(user_id)
-        if user is not None:
-            # Instead of replicating all that handling... gets attr from decorator
-            return await ctx.invoke(self.ban, user, None, reason=reason)
-        user = discord.Object(id=user_id)  # User not in the guild, but
+        for user_id in user_ids:
+            user = guild.get_member(user_id)
+            if user is not None:
+                # Instead of replicating all that handling... gets attr from decorator
+                try:
+                    await ctx.invoke(self.ban, user, days, reason=reason)
```

Comment:
Edge case still to be fixed here: at the moment if the invoked ban fails, the banning is still successful as far as the hackban command is concerned.